examples/create_custom_sbom/main.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"context"
+	"crypto"
+	"fmt"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/anchore/syft/syft"
+	"github.com/anchore/syft/syft/cataloging"
+	"github.com/anchore/syft/syft/cataloging/filecataloging"
+	"github.com/anchore/syft/syft/cataloging/pkgcataloging"
+	"github.com/anchore/syft/syft/file"
+	"github.com/anchore/syft/syft/sbom"
+	"github.com/anchore/syft/syft/source"
+)
+
+func main() {
+	// automagically get a source.Source for arbitrary string input
+	src := getSource("alpine:3.19")
+
+	// will catalog the given source and return a SBOM keeping in mind several configurable options
+	sbom := getSBOM(src)
+
+	// show a simple package summary
+	summarize(sbom)
+
+	// show the alpine-configuration cataloger results
+	showAlpineConfiguration(sbom)
+}
+
+func getSource(input string) source.Source {
+	// refactor: source.Detect should take pointer? (to allow for nil default)
+	// refactor: keith has suggestions for refactoring the source.Detection flow
+	detection, err := source.Detect(input, source.DetectConfig{
+		// refactor: this is a magic string
+		DefaultImageSource: "docker",
+	})
+
+	if err != nil {
+		panic(err)
+	}
+
+	// refactor: take pointer and allow for nil?
+	src, err := detection.NewSource(source.DetectionSourceConfig{})
```

Comment:
we should have used the default function https://github.com/anchore/syft/blob/c5d15d1d6c2e79f92545cf8e973317deb04b7638/syft/source/detection.go#L93